src/lib.rs

```diff
@@ -232,6 +226,66 @@ impl Options<'_> {
     }
 }
 
+/// The error returned by [`cmark_resume_one_event`] and
+/// [`cmark_resume_with_source_range_and_options`].
+#[derive(Debug)]
+pub enum Error {
+    FormatFailed(fmt::Error),
+    UnexpectedEvent,
+}
+
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::FormatFailed(e) => e.fmt(f),
+            Self::UnexpectedEvent => f.write_str("Unexpected event while reconstructing Markdown"),
+        }
+    }
+}
+
+impl std::error::Error for Error {}
+
+impl From<fmt::Error> for Error {
+    fn from(e: fmt::Error) -> Self {
+        Self::FormatFailed(e)
+    }
+}
+
+//======================================
```

Comment:
Could all labels (like these) be removed?
If further structuring is required, functions can be moved into their own modules that are named accordingly.

Reply:
Of course 🙂 That's a bit of my own personal style leaking through a bit too much; no problem with removing these :)

I just pushed a commit that drops these. Let me know if there's anything else! :)